src/lib.rs

```diff
@@ -156,132 +159,253 @@ impl Datadog {
         }
     }
 
-    /// initialize a Datadog instance with bare parameters. In this case reporting is enabled by default
-    /// This should be used carefully. Consider using [Datadog::init] instead
-    pub fn new(client: impl 'static + DogstatsdClient + Send + Sync) -> Self {
+    /// initialize a Datadog instance with bare parameters. This should be used carefully.
+    /// Consider using [Datadog::init] instead
+    pub fn new(client: impl 'static + DogstatsdClient + Send + Sync, is_reporting_enabled: bool) -> Self {
```

Comment:
Why is this still needed? can't we just make it private?

Reply:
Ok, We can do it public only in the crate

Reply:
maybe we use this just for testing purpose? So maybe we can conditionally compile it only in test env

Reply:
We need this because of [this](https://github.com/primait/prima_datadog.rs/pull/13#discussion_r877158993)